Remove usage of std::binary_function in the function since it's been long
deprecated and is beginning to cause problems with newer compilers

https://github.com/kismetwireless/kismet/commit/5bae9c68ce45940e03fda7d8abbbfda0fca9e95c
--- alphanum.hpp
+++ alphanum.hpp
@@ -304,7 +304,7 @@ namespace doj
      implement "std::ostream operator<< (std::ostream&, const Ty&)".
   */
   template<class Ty>
-  struct alphanum_less : public std::binary_function<Ty, Ty, bool>
+  struct alphanum_less 
   {
     bool operator()(const Ty& left, const Ty& right) const
     {
